c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+	const int Year = ValidateInputAndReturnValidatedInput(TemporalYear, "Please insert the actual year the game was published: ", 1900, 2022);
+
+	float TemporalMonth;
+	std::cout << "Please insert the month the game was published (1-12): ";
+	std::cin >> TemporalMonth;
+	int Month = ValidateInputAndReturnValidatedInput(TemporalMonth, "Please insert the actual month the game was published (1-12): ", 1, 12);
+
+	float TemporalDay;
+	std::cout << "Please insert the day the game was published: ";
+	std::cin >> TemporalDay;
+	int Day = GetValidDay(TemporalDay, Month, Year);
+
+	FVideogame Game = FVideogame(GameName, GameStudio, FDate(Day, Month, Year));
+
+	if (CategoryIndex == -1)
+	{
+		Uncategorized.AddGame(Game);
+	}
+
+	else
+	{
+		CategoryContainer.AddGameToCategory(CategoryIndex, Game);
+	}
+
+	NumberOfGames++;
+
+	ActiveCommand = EMenuCommand::MainMenu;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenCategoryManagerMenu()
+{
+	system("cls");
+	std::string CategoryManagerMessage = "What would you rather do next? \n1 - Add a new category\n2 - Remove a category\n3 - Go back to main menu\nTELL ME : ";
+	std::cout << CategoryManagerMessage;
+
+	enum class ECategoryManagerCommand { CreateCategory = 1, RemoveCategory, MainMenu };
+	ECategoryManagerCommand UserChoice;
+
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	UserChoice = (ECategoryManagerCommand)ValidateInputAndReturnValidatedInput(TemporalUserChoice, CategoryManagerMessage, 1, 3);
+
+	switch (UserChoice)
+	{
+	case ECategoryManagerCommand::CreateCategory:
+		OpenCreateCategory();
+		break;
+
+	case ECategoryManagerCommand::RemoveCategory:
+		OpenRemoveCategory();
+		break;
+
+	case ECategoryManagerCommand::MainMenu:
+		ActiveCommand = EMenuCommand::MainMenu;
+		return;
+
+	default:
+		std::cout << "CategoryManager Error, please contact support";
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenGameDisplayerMenu() {
+	system("cls");
+
+	if (NumberOfGames == 0)
+	{
+		std::cout << "There are currently no games to display. Please type Enter to return to the main menu. ";
+
+		std::cin.ignore(500, '\n');
+		if (std::cin.peek() == '\n')
+		{
+			OpenMainMenu();
+		}
+		return;
+	}
+
+	if (Uncategorized.GetCurrentNumberOfGames() != 0)
+	{
+		std::cout << "---------------------------------------------------------\n" << "Uncategorized Games\n" << std::endl;
+		for (int i = 0; i < Uncategorized.GetCurrentNumberOfGames(); i++)
+		{
+			FVideogame Game = Uncategorized.GetGame(i);
+			std::cout << Game.GetName() << "\t by " << Game.GetStudio() << "\t published on " << Game.GetFormattedDate() << std::endl;
+		}
+	}
+
+	if (!CategoryContainer.IsEmpty())
```

Comment:
This check is unnecessary here. Your loop won't run anything if the category container is empty, so this is just decreasing legibility by adding an extra nested block of code.